main/src/ui/java/de/blinkt/openvpn/fragments/PackageAdapter.kt

```diff
@@ -157,18 +179,35 @@ internal class PackageAdapter(c: Context, vp: VpnProfile) : RecyclerView.Adapter
         viewHolder.appIcon.setImageDrawable(mInfo.loadIcon(mPm))
         viewHolder.checkBox.tag = mInfo.packageName
         viewHolder.checkBox.setOnCheckedChangeListener(this)
-
+        viewHolder.checkBoxMobile.tag = mInfo.packageName
+        viewHolder.checkBoxMobile.setOnCheckedChangeListener(this)
 
         viewHolder.checkBox.isChecked = mProfile.mAllowedAppsVpn.contains(mInfo.packageName)
+        viewHolder.checkBoxMobile.isChecked = mProfile.mAllowedAppsVpnMobile.contains(mInfo.packageName)
+
+        if ( mProfile.mAllowAppMobile )
+            viewHolder.checkBoxMobile.visibility = CompoundButton.VISIBLE
+
+        else
+            viewHolder.checkBoxMobile.visibility = CompoundButton.GONE
     }
 
 
     override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
         val packageName = buttonView.tag as String
-        if (isChecked) {
-            mProfile.mAllowedAppsVpn.add(packageName)
-        } else {
-            mProfile.mAllowedAppsVpn.remove(packageName)
+        if ( buttonView.id == R.id.app_selected ) {
```

Comment:
The project uses the default android style that doesn't have extra space after ( in if